gammapy/irf/tests/test_background.py

```diff
@@ -21,35 +31,65 @@ def test_background_3d_basics(bkg_3d):
     assert 'NDDataArray summary info' in str(bkg_3d.data)
 
     axis = bkg_3d.data.axis('energy')
-    assert axis.nbins == 21
+    assert axis.nbins == 2
     assert axis.unit == 'TeV'
 
     axis = bkg_3d.data.axis('detx')
-    assert axis.nbins == 36
+    assert axis.nbins == 3
     assert axis.unit == 'deg'
 
     axis = bkg_3d.data.axis('dety')
-    assert axis.nbins == 36
+    assert axis.nbins == 3
     assert axis.unit == 'deg'
 
     data = bkg_3d.data.data
-    assert data.shape == (21, 36, 36)
+    assert data.shape == (2, 3, 3)
     assert data.unit == u.Unit('s-1 MeV-1 sr-1')
 
 
+def test_background_3d_read_write(tmpdir, bkg_3d):
+    filename = str(tmpdir / "bkg3d.fits")
+    bkg_3d.to_fits().writeto(filename)
+
+    bkg_3d_2 = Background3D.read(filename)
+
+    axis = bkg_3d_2.data.axis('energy')
+    assert axis.nbins == 2
+    assert axis.unit == 'TeV'
+
+    axis = bkg_3d_2.data.axis('detx')
+    assert axis.nbins == 3
+    assert axis.unit == 'deg'
+
+    axis = bkg_3d_2.data.axis('dety')
+    assert axis.nbins == 3
+    assert axis.unit == 'deg'
+
+    data = bkg_3d_2.data.data
+    assert data.shape == (2, 3, 3)
+    assert data.unit == 's-1 MeV-1 sr-1'
+
+
 @requires_dependency('scipy')
 @requires_data('gammapy-extra')
 def test_background_3d_evaluate(bkg_3d):
-    bkg_rate = bkg_3d.data.evaluate(energy='1 TeV', detx='0.2 deg', dety='0.5 deg')
-    assert_allclose(bkg_rate.value, 0.00013352689711418575)
-    assert bkg_rate.unit == 's-1 MeV-1 sr-1'
+    # Evaluate at nodes in energy
+    res = bkg_3d.evaluate(detx=np.array([1, 0.5]) * u.deg, dety=np.array([1, 0.5]) * u.deg,
```

Comment:
You can remove the `np.array` calls here and the result should be exactly the same, just a bit less code.